lib/modules/versioning/poetry/transform.ts

```diff
@@ -63,12 +63,18 @@ export function poetry2semver(
 
   const parts = [releaseParts.map((num) => num.toString()).join('.')];
   if (pre !== null) {
+    // trim leading zeros from valid numbers
+    pre.number = pre.number.replace(/^0+(?!$)/, '');
```

Comment:
no, too expensive regex.
```suggestion
    pre.number = pre.number.replace(regEx(/^0+\d/), '');
```
this is much cheaper because no look around

Reply:
it also doesn't work, it would drop the first digit, this is kind of why I expect that the [pre-existing code](https://github.com/renovatebot/renovate/blob/e0115f9bc4abde8047a4d40b38ccb6d8cdaebd71/lib/modules/versioning/poetry/transform.ts#L54) uses the `parseInt` way

`010` --> `0`

Reply:
okay I think I got it working with capture groups